File: server/src/UET.EasyAccommod.Core/Study/Book.cs
using Abp.Domain.Entities;
using Abp.Domain.Entities.Auditing;
using System.ComponentModel.DataAnnotations.Schema;

namespace UET.EasyAccommod.Study
{
    [Table("Book")]
    public class Book : FullAuditedEntity<long>, IEntity<long>
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public string BookColor { get; set; }
        public string BookImage { get; set; }
        public long Grade { get; set; }
        public long TotalWord { get; set; }
        public long TotalSentence { get; set; }
        public long TotalUnit { get; set; }
    }
}

File: mobile/lib/core/models/study/book/book.dart
import 'package:json_annotation/json_annotation.dart';
part 'book.g.dart';
@JsonSerializable(nullable: true, explicitToJson: true)
class ListBook{
  List<Book> items;
  ListBook({this.items});
  factory ListBook.fromJson(Map<String, dynamic> json) => _$ListBookFromJson(json);

  Map<String, dynamic> toJson() => _$ListBookToJson(this);
}
@JsonSerializable(nullable: true, explicitToJson: true)
class Book {
  String name;
  String description;
  String bookColor;
  String bookImage;
  int grade;
  int totalWord;
  int totalSentence;
  int totalUnit;
  int id;

  Book(
      {this.name,
        this.description,
        this.bookColor,
        this.bookImage,
        this.grade,
        this.totalWord,
        this.totalSentence,
        this.totalUnit,
        this.id});
  factory Book.fromJson(Map<String, dynamic> json) => _$BookFromJson(json);

  Map<String, dynamic> toJson() => _$BookToJson(this);
}